go.mod

```diff
@@ -1,165 +1,185 @@
 module github.com/openshift/oadp-operator
 
-go 1.20
+go 1.22.0
```

Comment:
I think we said this is no longer necessary if you got the 1.22.3 or above builder. The requirement for patch version was a bug.

Reply:
There are 2 options: specify the .0 here *or* add the toolchain line.

In other words, we have 2 choices:
```
go 1.22.0
```
or:
```
go 1.22

toolchain go1.22.3
```

I don't really have much of a preference. Note that if you remove ".0" and run `go mod tidy` the toolchain line is added. We need one or the other.

Reply:
I can run `go mod tidy` without toolchain being added (go version go1.22.3 linux/amd64)

But we can copy how is done in Velero go.mod

Reply:
I think upstream is gonna be 1.22.2 https://github.com/vmware-tanzu/velero/pull/7821/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

Reply:
@shubham-pampattiwar ok - -we'll get that on the next rebase for velero. so should we set `go 1.22.2` in our go.mod for oadp-operator then?

Reply:
@mateusoliveira43 for me, tidy added toolchain if I have `go 1.22` but not if I have `go 1.22.0` -- I'm also on 1.22.3.

Reply:
FYI, in case it matters, we don't have a downstream/internal golang-builder newer than `v1.22.1` yet

Reply:
@sseago yeah if upstream is gonna release velero 1.14 with Golang 1.22.2 then we should do the same for oadp-1.4 imo

Reply:
updated to 1.22.2